pandas/util/testing.py

```diff
@@ -1415,27 +1438,36 @@ def assert_equal(left, right, **kwargs):
     """
     __tracebackhide__ = True
 
-    if isinstance(left, pd.Index):
+    if isinstance(left, Index):
+        right = cast(Index, right)
```

Comment:
Shouldn't need any of the casts here; `isinstance` will narrow the types appropriately

Reply:
if I remove the cast I get the following error. 
```
pandas/util/testing.py:1440: error: Argument 2 to "assert_index_equal" has incompatible type "Union[DataFrame, Index]"; expected "Index"
pandas/util/testing.py:1440: error: Argument 2 to "assert_index_equal" has incompatible type "Union[DataFrame, Any]"; expected "Index"
```